.github/workflows/package-ubuntu.yml

```diff
@@ -0,0 +1,207 @@
+        run: |
+          wget https://apt.llvm.org/llvm.sh
+          chmod +x llvm.sh
+          ./llvm.sh ${{matrix.clang}}
+          apt install -y libclang-${{matrix.clang}}-dev clang-tools-${{matrix.clang}} libomp-${{matrix.clang}}-dev llvm-${{matrix.clang}}-dev liblld-${{matrix.clang}}-dev
+          apt install -y libstdc++-12-dev
+      - name: install dependencies
+        run: |
+          apt install -y libboost-all-dev libtbb-dev cmake python3
+      - name: install OpenCL development files and loader
+        run: |
+          apt install -y libopencl-clang-dev ocl-icd-opencl-dev
+      - name: setup build environment
+        run: |
+          export CXXFLAGS="$CXXFLAGS"
+          if [[ "${{matrix.clang}}" != "11" && "${{matrix.clang}}" -lt "16" ]]; then
+            export OMP_CXX_FLAGS="$CXXFLAGS -fexperimental-new-pass-manager"
+            export CC=clang-${{matrix.clang}}
+            export CXX=clang++-${{matrix.clang}}
+          fi
+          echo "CC=${CC}" >> $GITHUB_ENV
+          echo "CXX=${CXX}" >> $GITHUB_ENV
+          echo "OMP_CXX_FLAGS=${OMP_CXX_FLAGS}" >> $GITHUB_ENV
+      - name: install packaging helpers
+        run: |
+          apt install -y build-essential devscripts debhelper
+      - name: package AdaptiveCpp
+        run: |
+          cd ..
+          cp -R AdaptiveCpp acpp-24.02.0
+          tar -cf acpp_24.02.0.orig.tar.gz acpp-24.02.0/
+          cd acpp-24.02.0/
+          echo $(cat  /var/lib/dpkg/info/libllvm17\:amd64.shlibs) > debian/shlibs.local
+          debuild -us -uc
+      - name: Install the package
+        run: |
+          cd ${GITHUB_WORKSPACE}/../
+          dpkg -i acpp_24.02.0-1_amd64.deb
+      - name: Release Packages
+        uses: actions/upload-artifact@v4
+        with:
+          name: cpu-backend-generic
+          path: /home/runner/work/AdaptiveCpp/acpp_24.02.0-1_amd64.deb
+      - name: build generic SSCP tests
+        if: matrix.clang >= 14
+        run: |
+          mkdir ${GITHUB_WORKSPACE}/build-tests-sscp
+          cd ${GITHUB_WORKSPACE}/build-tests-sscp
+          cmake -DACPP_TARGETS="generic" -DAdaptiveCpp_DIR=${GITHUB_WORKSPACE}/build/install/lib/cmake/AdaptiveCpp -DWITH_PSTL_TESTS=ON ${GITHUB_WORKSPACE}/tests
+          make -j2
+      - name: build CPU tests
+        run: |
+          mkdir ${GITHUB_WORKSPACE}/build-tests-cpu
+          cd ${GITHUB_WORKSPACE}/build-tests-cpu
+          cmake -DACPP_TARGETS="omp" -DAdaptiveCpp_DIR=/usr/lib/cmake/AdaptiveCpp ${GITHUB_WORKSPACE}/tests
+          make -j2
+      - name: run CPU tests
+        run: |
+          cd ${GITHUB_WORKSPACE}/build-tests-cpu
+          LD_LIBRARY_PATH=/usr/lib ./sycl_tests
+  packaging-acpp-all-backends:
+    name: Packaging AdaptiveCpp, clang ${{matrix.clang}}, ${{matrix.os}}, CUDA ${{matrix.cuda}}, ROCm ${{matrix.rocm}}
+    runs-on: ${{matrix.os}}
+    strategy:
+      matrix:
+        clang: ['17']
+        os: [ubuntu-22.04]
+        cuda: [11.0.2]
+        rocm: [5.6.1]
```

Comment:
clang 17 is likely too new for ROCm 5.6.
SSCP requires that clang <= ROCm clang.
The old compilation flow requires that clang >= ROCm clang.

So best bet is probably to use the same clang/LLVM version that ROCm uses.

Reply:
Okay, then I'll downgrade clang to 16 since that's what's shipped in ROCm 5.6 👍